src/core/jit/request.rs

```diff
@@ -46,8 +46,8 @@ impl Request<ConstValue> {
         let plan = builder.build(self.operation_name.as_deref())?;
 
         transform::CheckConst::new()
+            .pipe(transform::AuthPlaner::new(&blueprint.server.auth))
```

Comment:
AuthPlanner goes through fields and if there's protected IR it set's that in query plan, right?
so we need to run this transformer only when `blueprint.server.auth.is_some()`.

we can utilise the when operator here to do that.